cai_causal_graph/graph_components.py

```diff
@@ -310,6 +325,45 @@ def __hash__(self) -> int:
         """Return a hash value of the node identifier."""
         return super().__hash__()
 
+    def to_dict(self, include_meta: bool = True) -> dict:
+        """
+        Return a dictionary representation of the time series node.
+
+        :param include_meta: Whether to include the metadata in the dictionary. Default is `True`.
+        :return: The dictionary representation of the time series node.
+        """
+        dictionary = super().to_dict(include_meta)
+        # add the time lag and variable name to the dictionary
+        dictionary.update({TIME_LAG: self.time_lag, VARIABLE_NAME: self.variable_name})
+        return dictionary
+
+    @classmethod
+    def from_dict(cls, dictionary: dict) -> TimeSeriesNode:
+        """
+        Create a time series node from a dictionary.
```

Comment:
Follow the example for `from_dict` for `Node` and `Edge` which includes full path to the classes.